packages/block-library/src/navigation-link/edit.js

```diff
@@ -193,6 +215,11 @@ function NavigationLinkEdit( {
 					/>
 				</PanelBody>
 			</InspectorControls>
+			<BlockNavigationListItemFill blockId={ clientId }>
```

Comment:
You may be able to avoid the need to pass the `clientId` here by using something like `BlockContext` internally in the `ListItemSlot`, this seems to have the `clientId`:
https://github.com/WordPress/gutenberg/blob/6c26cdf1f97307b478e37d731351c4b1125e902b/packages/block-editor/src/components/block-list/block-wrapper.js#L36-L53

Or there might be another way. I think all the other slots rely on showing only content for the selected block, whereas this use case is a bit different.